### ISSUE TITLE
Duplicate cyborg ghost after being rebuilt

### ISSUE THREAD

**Issue by Saicchi** [goonstation/goonstation#11219]
### Labels

[BUG]

### Description

Duplicate cyborg ghost after being rebuilt

### Steps to reproduce

1.  Join as a late game cyborg.
2. Unlock your interface and remove your own head ( crowbar, screwdriver, wrench ).
3. Rebuild the same cyborg with the same parts and activate it.
4. Unlock the interface and remove the brain ( screwdriver )
5. You are now dead and should be seeing a duplicate of your own ghost.

### Expected Behavior

No duplicate ghost.

### Additional Information & Screenshots

![image](https://user-images.githubusercontent.com/47158232/194420401-2274c110-681f-4a48-b831-87e0f76314bd.png)

Me as `C.Z.D` the name I picked when I joined late-game and `Cyborg Pi-29` which is the ghost I could control after step 4.
Unknown if this can be tested with only a single person, might need two.

**Comment by TobleroneSwordfish** [comment]
pretty sure this is just how ghosts work, not exactly a bug

**Comment by Gerhazo** [comment]
Ghosts are attached to mobs rather than clients/players, rebuilding the cyborg presumably creates a new mob, resulting in a new ghost.

That's not to say this isn't a bug: ghosts are generally hidden when uninhabited and there could be a case made for cleaning up ghosts when a mob is deleted and there isn't any player associated with either the ghost or the body, so this probably is unintended behaviour.
